dartagnan/src/main/java/com/dat3m/dartagnan/program/event/pthread/Create.java

```diff
@@ -53,6 +54,8 @@ public Create getCopy(){
 
     @Override
     protected RecursiveFunction<Integer> compileRecursive(Arch target, int nextId, Event predecessor, int depth) {
+    	Preconditions.checkArgument(target != null, "Target cannot be null");
```

Comment:
Since you changed the error message, a `checkNotNull` is more appropriate. When I suggested `checkArgument` it was with respect to the old message.
I think the cleanest way is to always do `checkNotNull` and if some event can only be compiled to specific targets, there should additionally be a `checkArgument`.